### PR TITLE
Add support for product/serialno config for recovery

### PR DESCRIPTION
**Pull request by althafvly**

### CODE REVIEW

File: META-INF/com/google/android/update-binary [hunk 1]
```diff
@@ -120,8 +120,25 @@ if [ "$SYSTEM_STORAGE" -lt "$NEEDED_STORAGE_SYSTEM" ]; then
...
+  CONFIG=$DEFAULT_CONFIG.$SERIALNO
+elif [[ `unzip -l $ZIP | grep ih8sn.conf.$PRODUCT` ]] && [[ ! -z "$PRODUCT" ]]; then
+  CONFIG=$DEFAULT_CONFIG.$PRODUCT
+else
+  CONFIG=$DEFAULT_CONFIG
+fi
+
 ui_print "Copying files"
-unzip -qq "$ZIP" system/etc/init/ih8sn.rc system/etc/ih8sn.conf system/addon.d/60-ih8sn.sh system/bin/ih8sn -d "${SYSTEM_MNT}/"
+unzip -qq "$ZIP" system/etc/init/ih8sn.rc $CONFIG system/addon.d/60-ih8sn.sh system/bin/ih8sn -d "${SYSTEM_MNT}/"
```
Comment (luk1337) [inline META-INF/com/google/android/update-binary:136]: maybe we could just run another unzip for config, e.g.:
```
unzip -qqp "$ZIP" $CONFIG > "${SYSTEM_MNT}/system/etc/ih8sn.conf"
```

**Comment by althafvly** [comment]
done

File: META-INF/com/google/android/update-binary [hunk 2]
```diff
@@ -120,8 +120,22 @@ if [ "$SYSTEM_STORAGE" -lt "$NEEDED_STORAGE_SYSTEM" ]; then
   fi
 fi
 
+# Set ih8sn config
+SERIALNO=$(getprop ro.boot.serialno)
+PRODUCT=$(getprop ro.build.product)
+DEFAULT_CONFIG=system/etc/ih8sn.conf
+if [[ `unzip -l $ZIP | grep ih8sn.conf.$SERIALNO` ]] && [[ ! -z "$SERIALNO" ]]; then
```
Comment (luk1337) [inline META-INF/com/google/android/update-binary:127]: `ih8sn.conf.$SERIALNO$`

File: META-INF/com/google/android/update-binary [hunk 3]
```diff
@@ -120,8 +120,22 @@ if [ "$SYSTEM_STORAGE" -lt "$NEEDED_STORAGE_SYSTEM" ]; then
...
+  CONFIG=$DEFAULT_CONFIG.$SERIALNO
+elif [[ `unzip -l $ZIP | grep ih8sn.conf.$PRODUCT` ]] && [[ ! -z "$PRODUCT" ]]; then
```
Comment (luk1337) [inline META-INF/com/google/android/update-binary:129]: `ih8sn.conf.$PRODUCT$`

File: META-INF/com/google/android/update-binary (hunk 2 above)
Comment (althafvly) [reply META-INF/com/google/android/update-binary]: done

**Pull request merged**